tests/test_butler.py

```diff
@@ -1914,17 +1922,31 @@ def assertButlerTransfers(self, id_gen_map=None, purge=False, storageClassName="
                     # For one of these delete the file as well.
                     # This allows the "missing" code to filter the
                     # file out.
+                    # Access the individual datastores.
+                    datastores = []
+                    if hasattr(butler.datastore, "datastores"):
+                        datastores.extend(butler.datastore.datastores)
+                    else:
+                        datastores.append(butler.datastore)
+
                     if not deleted:
-                        primary, uris = butler.datastore.getURIs(ref)
-                        if primary:
-                            primary.remove()
-                        for uri in uris.values():
-                            uri.remove()
+                        # For a chained datastore we need to remove
+                        # files in each chain.
+                        for datastore in datastores:
+                            primary, uris = datastore.getURIs(ref)
+                            if primary:
+                                if primary.scheme != "mem":
+                                    primary.remove()
```

Comment:
Maybe the check for "mem" is better done inside `remove()` so that you need to know less about this special case?

Reply:
Hmm. Yes, I can probably make `resource.remove()` a no-op for `mem://` -- I think I resisted doing that because you could conceive of people thinking that would delete the in-memory resource when it can't actually do that.